src/controllers/dynakube/version/updater.go

```diff
@@ -130,16 +146,14 @@ func setImageIDWithDigest( //nolint:revive
 				target.ImageID = canonRef.String()
 			}
 		}
+		target.Version = imageVersion.Version
```

Comment:
if you don't set the `target.Version` in the `canonRef, ok := ref.(reference.Canonical); ok` case,  then you will error (this can happen in the `customImage` case)

you can just set the version to the digest in that case, which is a bit scary as `CheckForDowngrade` would go boom as a digest is not a semantic version, but we don't do that in the case of `customImage` so it should be fine 😬

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/2156/commits/5c6a5fff8d2dfd2c794118618de21306aaf2f9e7